### PR TITLE
Add valid status codes assertion by default using the defined responses

### PR DESCRIPTION
**Pull request by prakashmrec**

### CODE REVIEW

File: src/main/groovy/com/smartbear/swagger/assertion/ValidStatusCodesAssertionCreator.groovy [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import io.swagger.models.Response
+
+import static com.eviware.soapui.security.assertion.ValidHttpStatusCodesAssertion.LABEL
+
+
+class ValidStatusCodesAssertionCreator implements AssertionCreator {
+    @Override
+    void createAssertion(RestTestRequestStep restTestRequestStep, Map<String, Response> responseMap) {
+        ValidHttpStatusCodesAssertion assertion = (ValidHttpStatusCodesAssertion) restTestRequestStep.addAssertion(LABEL)
+        assertion.codes = responseMap.keySet().join(',')
```
Comment (olensmar) [inline src/main/groovy/com/smartbear/swagger/assertion/ValidStatusCodesAssertionCreator.groovy]: I think Swagger has the concept of a "default" response code - we need to filter these out - see http://swagger.io/specification/#responsesDefault

Comment (olensmar) [inline src/main/groovy/com/smartbear/swagger/assertion/ValidStatusCodesAssertionCreator.groovy]: Also - judging from http://swagger.io/specification/#headersObject - we should be able to add assertions that validate that the specified headers exist in the response!?

Comment (prakashmrec) [inline src/main/groovy/com/smartbear/swagger/assertion/ValidStatusCodesAssertionCreator.groovy]: As discussed offline - we will skip the ValidStatusCodesAssertion if there is a default response specified. 
Headers are not mandatory - it just provided a list of possible headers which could be present in reponse, hence we can't add assertion on them.

**Comment by olensmar** [comment]
did we talk about adding the SwaggerComplianceAssertion also?

**Comment by prakashmrec** [comment]
Yes, but that will be in a separate PR.

**Comment by olensmar** [comment]
mighty fine!

**Pull request merged**